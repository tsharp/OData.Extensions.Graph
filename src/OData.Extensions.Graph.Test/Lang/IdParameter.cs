using HotChocolate.Language;
using OData.Extensions.Graph.Lang;
using Snapshooter.Xunit;

namespace OData.Extensions.Graph.Test.Lang
{
    public class IdParameter
    {
        // [Fact]
        public static void SimpleIdParam()
        {
            var query = "{user(Id: { eq: \"abc\" }) { items { Id } } }";

            Utf8GraphQLParser.Parse(query).ToString(true).MatchSnapshot();

            // Arrange
            var translator = new OperationTranslator(DebugBindingResolver.Instance, Common.GetEdmModel());

            // Act
            var filerByUserId = translator.TranslateQuery("/user('abc')?$select=Id");

            // Assert
            filerByUserId.DocumentNode.ToString(true).MatchSnapshot();
        }
    }
}
