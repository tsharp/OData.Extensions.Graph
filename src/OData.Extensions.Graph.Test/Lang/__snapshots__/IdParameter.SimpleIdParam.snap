{
  user(Id: { eq: "abc" }) {
    items {
      Id
    }
  }
}
